trace/config.go

```diff
@@ -203,3 +205,56 @@ func (i instrumentationVersionOption) ApplyTracer(config *TracerConfig) {
 }
 
 func (instrumentationVersionOption) private() {}
+
+// ErrorConfig is a group of options for error.
+type ErrorConfig struct {
+	Code    codes.Code
+	Message string
+	// EventOpts will be passed to addEvent
+	EventOpts []EventOption
+}
+
+// ErrorOption applies an option to a ErrorConfig.
+type ErrorOption interface {
+	ApplyError(config *ErrorConfig)
```

Comment:
```suggestion
	applyError(config *errorConfig)
```

We're working on [normalizing configuration and options](https://github.com/open-telemetry/opentelemetry-go/pull/1882/files) and part of that will involve making the config type and apply methods unexported.  Can we get out ahead of that here?

Reply:
Sure. I had renamed  ApplyError to applyError.
But golangci-lint will  fail if I changed  ErrorConfig to errorConfig. 

`
exported func NewErrorConfig returns unexported type *go.opentelemetry.io/otel/trace.errorConfig, which can be annoying to us.
`

Reply:
`newErrorConfig` shouldn't need to be exported, either.

Reply:
Since newErrorConfig will be used in by bridge/opentracing/internal/mock.go ,  oteltest/span.go and sdk/trace/span.go, it will take a lot of work to make it unexported at this time.

I think it's better to move all the code which implements Span interface to the package trace, so we can easily make the config unexported.

But there is another problem,  since  sdk/trace/span.go needs some code in  sdk/trace/internal, we must change it at the same time.

I'm not sure about this solution, it will cause a lot of files to be changed.

Maybe we can keep NewErrorConfig exported and  make a new pr to make it unexported, I'd like to take it.